[GEOS-10113] Geofence REST NPE when requesting rule with void Layer details - [GEOS-10114] Geofence server UI, error when changing the layerType from layer details tab

taba90 opened the pull request:
[![GEOS-10113](https://badgen.net/badge/JIRA/GEOS-10113/0052CC)](https://osgeo-org.atlassian.net/browse/GEOS-10113)<!-- PR-BADGE: PLEASE DO NOT REMOVE THIS COMMENT -->

The pr fixes two bugs:
[GEOS-10113](https://osgeo-org.atlassian.net/browse/GEOS-10113) NPE from get rest service endpoints if a rule has null layerType.
[GEOS-10114](https://osgeo-org.atlassian.net/browse/GEOS-10114) missing outputmarkup id on some UI components. The related commit make also the spatial filter type behavior consistent among tabs and fix a test that was not working properly(although successful the form setValue calls were not setting anything).
## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**


For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

aaime commented:
The code changes look legit. I've tried fiddling with the GUI, but cannot find a way to change things in the layer type drop down, it's always disabled and its content seem to be determined by the layer chosen in the first tab?

taba90 commented:
@aaime 
according to our discussion about the LayerType dropdown in the LayerDetails tab UI:
- I've removed it since layerType should be handle automatically from geoserver, and replaced with a simple label.
- Modified the tests accordingly

aaime commented on src/extension/geofence-server/src/main/java/org/geoserver/geofence/web/GeofenceRulePage.java:
```diff
@@ -944,6 +951,24 @@ protected Component getComponentForProperty(
         }
     }
 
+    private boolean isEnableFilterAndStyles(LayerDetailsFormData detailsFormData) {
```
```suggestion
    private boolean isFilterAndStylesEnabled(LayerDetailsFormData detailsFormData) {
```

taba90 replied:
fixed

aaime commented on src/extension/geofence-server/src/main/java/org/geoserver/geofence/web/GeofenceRulePage.java:
```diff
@@ -944,6 +951,24 @@ protected Component getComponentForProperty(
+        return isEnableFilterAndStyles(detailsFormData.layerType);
+    }
+
+    private boolean isEnableFilterAndStyles(LayerType layerType) {
```
Same as above.

taba90 replied:
fixed

aaime reviewed (changes requested):
Looks good, noted a few small improvements.
